ENIGMAsystem/SHELL/Platforms/General/UNIXfilemanip.cpp

```diff
@@ -20,48 +20,626 @@
 #include <unistd.h>
 #include <sys/stat.h>
 
+#include <stdio.h>
+#include <libgen.h>
+#include <dirent.h>
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+
+#include <fts.h>
+#include <stdlib.h>
+#include <limits.h>
+
 #include <string>
 #include "PFfilemanip.h"
 using namespace std;
 
 /* UNIX-ready port of file manipulation */
 
+struct dupslash
+{
+    bool operator()(char a, char b) const
+    {
+        return (a == '/' && b == '/');
+    }
+};
+
 namespace enigma_user
 {
 
-int file_exists(string fname)
+int file_exists(std::string fname)
 {
-  struct stat st;
-  return (stat(fname.c_str(),&st) == 0) and !(S_ISDIR(st.st_mode));
+    std::string str(fname);
+
+    struct stat sb;
+    return (stat((char *)str.c_str(),&sb) == 0 &&
+        S_ISREG(sb.st_mode));
+}
+
+int file_delete(std::string fname)
+{
+    std::string str(fname);
+
+    struct stat sb;
+    if (stat((char *)str.c_str(),&sb) == 0 &&
```

Comment:
Why are you doing a stat on the file before deleting it? The `remove` method will set `errno` to tell you why the remove failed. But since you're returning false if the file doesn't exist, anyway, I'm not sure why you're bothering to check. This entire method is equivalent to `return remove(fname.c_str()) == 0;`.